src/main/java/com/minecolonies/coremod/colony/jobs/JobDeliveryman.java

```diff
@@ -170,15 +170,15 @@ public void addRequest(@NotNull final IToken token)
      *
      * @param successful True when the processing was successful, false when not.
      */
-    public void finishRequest(@NotNull final boolean successful)
+    public void finishRequest(final boolean successful)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "successful" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)